xbmc/music/windows/GUIWindowMusicNav.cpp

```diff
@@ -287,6 +287,10 @@ bool CGUIWindowMusicNav::GetDirectory(const std::string &strDirectory, CFileItem
       OnRetrieveMusicInfo(items);
   }
 
+  //Navigating music files so default content to "songs" unless in sources folder.
+  //This content allows view type to include media info so that file tag data can be displayed
+  if (!URIUtils::IsSourcesPath(strDirectory))
```

Comment:
Does http://pastebin.com/VA7bryQk make it better? It only sets the content to "songs" if all other special cases don't apply and if it's neither the virtual directory, a `library://` path nor the source path.

I tried to compare it to the current behaviour but it looks exaclty the same and I'm not sure what the original issue was because I never use music files view.